src/generators/dart.ts

```diff
@@ -204,35 +214,45 @@ export function _toDart(requests: Request[], warnings: Warnings = []): string {
           "    ..fields[" + name + "] = " + repr(m.content, imports) + "\n";
       }
     }
+    multipart += "    ;\n\n";
 
-    if (hasHeaders || request.urls[0].auth) {
-      s += "  var req = new " + multipart;
-      for (const [hname, hval] of request.headers) {
-        s +=
-          "  req.headers[" +
-          repr(hname, imports) +
-          "] = " +
-          repr(hval || new Word(), imports) +
-          ";\n";
+    if (hasHeaders || request.urls[0].auth || rawRequestObj) {
```

Comment:
```suggestion
    if (hasHeaders || request.urls[0].auth) {
```

we're in an `if` block with this boolean (line 142 old/152 new)